include/karabo-bridge/kb_client.hpp

```diff
@@ -629,11 +629,33 @@ class Client {
     }
 
 public:
-    Client(): ctx_(1), socket_(ctx_, ZMQ_REQ) {}
+    /*
+     * Constructor.
+     *
+     * @param timeout: connection timeout in milliseconds. "-1" (default) for infinite.
+     */
+    Client(int timeout=-1): ctx_(1), socket_(ctx_, ZMQ_REQ) {
+      socket_.setsockopt(ZMQ_RCVTIMEO, timeout);
+      socket_.setsockopt(ZMQ_LINGER, 0);
+    }
+
+    // The destructor of zmq::context_t calls 'zmq_ctx_destroy'.
+    // The destructor of zmq::socket_t calls 'zmq_close'.
+    ~Client() = default;
+
+    // The copy and copy assignment constructor are implicitly deleted since
+    // those of zmq::context_t and zmq::socket_t are deleted.
+    Client(const Client&) = delete;
+    Client& operator=(const Client&) = delete;
 
     void connect(const std::string& endpoint) {
         std::cout << "Connecting to server: " << endpoint << std::endl;
-        socket_.connect(endpoint.c_str());
+        socket_.connect(endpoint);
+    }
+
+    void close() {
```

Comment:
I am sure whether we need this one. I am using it to shutdown a client with infinite timeout in the unittest. It causes SIGABRT roughly 1 out 10 runs. But I think creating and using `socket` in different threads is not encouraged. I need to brush up my knowledge about zmq.

`pfd.revents & POLLIN (src/signaler.cpp:264)`